analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpMetrics.cs

```diff
@@ -161,8 +162,13 @@ protected override bool IsStatement(SyntaxNode node)
                 case SyntaxKind.YieldReturnStatement:
                     return true;
 
-                default:
+                case SyntaxKind.Block:
                     return false;
+
+                default:
```

Comment:
This will help us catch if we don`t support some syntax, on the other side it may create additional issues for the users.

Reply:
Indeed. The purpose is to fail as fast as possible to know when we don't compute the metrics correctly. This is also a way to check if our safety net is good enough or if we have to improve it.